### DISCUSSION TITLE
Error: Node.js 12 actions are deprecated

### DISCUSSION THREAD

**Discussion by JoelMTaylor** [retypeapp/retype#569] [General]
Good day,

Thank you for your continued efforts to create an amazing product. I just saw the update to v3 - it looks great.

I am getting an error that prevents Regenerate Pages and Publish to ___ Directory.
![image](https://github.com/retypeapp/retype/assets/97919641/703dc49e-6609-43ac-9aa3-7a993599d67a)

The error text is: "Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: actions/checkout@v2. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.".

Drilling-down to the error shows: 
![image](https://github.com/retypeapp/retype/assets/97919641/a5ed7533-aafb-49f9-93b0-fdf12fbbeda3)

There have been similar errors recently. However, about two/three weeks ago, there was a different issue.
![image](https://github.com/retypeapp/retype/assets/97919641/d53d514c-3036-4317-9a00-89f975bcb239)
![image](https://github.com/retypeapp/retype/assets/97919641/c105d5b6-d721-4641-9478-8b7e2d3ba564)

I'm not sure why these issues are appearing now. No changes have been made to the actions (or anything else important) - just adding additional ```md``` pages. Any assistance would be greatly appreciated.

@geoffreymcgill, you still have access to our repository (https://github.com/ccmltd/docs-corportatesystems/actions).

**Comment by geoffreymcgill** [comment]
Likely the license key has expired. The original term was for 1 year and then a renewal was required, but we don't have an automated renewal system in place yet. Please send an email to hello@retype.com and we will send you a new extended license key.

**Reply by JoelMTaylor** [reply]
Thanks, we will do this.

**Comment by geoffreymcgill** [comment]
We need to improve this ERROR message, so that it's clear what exactly is going wrong.

**Reply by JoelMTaylor** [reply]
So Node.js 12 isn't an issue? 

I don't know much about how Retype would work, but I suppose it is built on Node.js and is still using v12? This is a vague question, what I really would like to know is: _is there anything I need to do (or can do) to fix this v12/v16 'error' - now or in the future?_

**Reply by geoffreymcgill** [reply]
I am now seeing the same `Node.js 12 actions are deprecated.` warning on the retype.com build, but they are just Warnings and do not stop the build. We are investigating and will update the build process to correct the warnings.

https://github.com/retypeapp/retype/actions/runs/5203951995

Your builds are failing because the license key is expired.

**Comment by geoffreymcgill** [comment]
The `Node.js 12 actions are deprecated.` warning can be fixed by updating your GitHub `actions/checkout` action to use `v3` instead of `v2`. This is a change GitHub made and not specific to anything required in Retype.

```sh
# Existing
- uses: actions/checkout@v2

# Revised
- uses: actions/checkout@v3
```

https://github.com/ccmltd/docs-corportatesystems/blob/ba5fe94e66a822dbc42a21dfaa39a1f576a19252/.github/workflows/retype-action.yml#L19

I also see that you use `actions/checkout@v2` on line `#37`, although I'm not sure if that `build_and_deploy_job` is related to Retype. You should likely update that line to use `actions/checkout@v3` as well, but test to ensure it works before pushing to production. 

https://github.com/ccmltd/docs-corportatesystems/blob/ba5fe94e66a822dbc42a21dfaa39a1f576a19252/.github/workflows/retype-action.yml#L37

We have updated the sample `retype-action.yml` in the Retype docs to reflect this change to `actions/checkout@v3`, see 
https://retype.com/guides/github-actions/#step-1-add-retype-actionyml-workflow

We have a fix in place for the `The set-output command is deprecated and will be disabled soon.` warning. I will post an update once testing is complete. No changes to your project will be required to fix this warning.

Hope this helps.

**Reply by geoffreymcgill** [reply]
The `The set-output command is deprecated and will be disabled soon.` warning has been fix. No changes to your project are required. Just push a change to your GitHub repo and the project will be rebuilt using the latest Retype **action-build** release. 

Here is the latest change to the retype.com project, and no `warning` messages during the workflow run:

https://github.com/retypeapp/retype/actions/runs/5247443011

Hope this helps.
